src/app/shell/item-comparators.ts

```diff
@@ -150,7 +150,21 @@ const ITEM_COMPARATORS: {
   // new -> old
   acquisitionRecency: acquisitionRecencyComparator,
   // None -> Kinetic -> Arc -> Thermal -> Void -> Raid -> Stasis
-  element: compareBy((item) => item.element?.enumValue ?? Number.MAX_SAFE_INTEGER),
+  elementWeapon: compareBy((item) => {
+    if (item.itemCategoryHashes.includes(ItemCategoryHashes.Weapon)) {
```

Comment:
`item.bucket.inWeapons` is a pre-computed way to do this check for DimItems

Reply:
Thanks. I was looking for something like that but must have missed it somehow.